distributor/pkg/lib/uniformlog.go

```diff
@@ -0,0 +1,103 @@
+package lib
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	cloudevents "github.com/cloudevents/sdk-go/v2"
+	keptnapimodels "github.com/keptn/go-utils/pkg/api/models"
+	keptn "github.com/keptn/go-utils/pkg/api/utils"
+	keptnv2 "github.com/keptn/go-utils/pkg/lib/v0_2_0"
+	logger "github.com/sirupsen/logrus"
+	"strings"
+)
+
+type UniformLog interface {
+	Start(ctx context.Context, eventChannel chan cloudevents.Event)
+}
+
+type EventUniformLog struct {
+	IntegrationID string
+	logHandler    keptn.ILogHandler
+}
+
+func NewEventUniformLog(integrationID string, logHandler keptn.ILogHandler) *EventUniformLog {
+	return &EventUniformLog{
+		IntegrationID: integrationID,
+		logHandler:    logHandler,
+	}
+}
+
+func (l *EventUniformLog) Start(ctx context.Context, eventChannel chan cloudevents.Event) {
+	logger.Info("starting UniformLog")
+	l.logHandler.Start(ctx)
+	go func() {
+		for {
+			select {
+			case event := <-eventChannel:
+				logger.Infof("UniformLogger: received event: %s", event.Context.GetType())
+				if err := l.OnEvent(event); err != nil {
+					logger.Errorf("could not handle event: %s", err.Error())
+				}
+			case <-ctx.Done():
+				logger.Info("closing UniformLogger")
+			}
+
+		}
+	}()
+}
+
+func (l *EventUniformLog) OnEvent(event cloudevents.Event) error {
+	keptnEvent, err := keptnv2.ToKeptnEvent(event)
+	if err != nil {
+		return fmt.Errorf("could not decode CloudEvent to Keptn event: %v", err.Error())
+	}
+	if keptnEvent.Type == nil {
+		return errors.New("received Keptn Event has no type")
+	}
+	if strings.HasSuffix(*keptnEvent.Type, ".finished") {
```

Comment:
should we introduce a func in go-utils for checking if an event is a finished event?